File: desktop/src/main/java/com/exodus/game/DesktopLauncher.java
package com.exodus.game;

import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application;
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3ApplicationConfiguration;
import com.badlogic.gdx.graphics.glutils.HdpiMode;

public class DesktopLauncher {
    public static void main(String[] args) {
        if (StartupHelper.startNewJvmIfRequired()) return; // Handles -XstartOnFirstThread
        
        Lwjgl3ApplicationConfiguration config = new Lwjgl3ApplicationConfiguration();
        
        // Window settings
        config.setTitle("Exodus Game");
        config.setWindowedMode(1280, 720);
        config.useVsync(true);
        config.setForegroundFPS(60);
        config.setBackBufferConfig(8, 8, 8, 8, 16, 0, 4);
        
        // HiDPI support
        config.setHdpiMode(HdpiMode.Logical);
        
        // OpenGL settings
        config.setOpenGLEmulation(Lwjgl3ApplicationConfiguration.GLEmulation.GL20, 2, 0);
        config.enableGLDebugOutput(true, System.out);
        
        // Window icon - TODO: Add proper icons
        //config.setWindowIcon("icon128.png", "icon64.png", "icon32.png", "icon16.png");
        
        new Lwjgl3Application(new ExodusGame(), config);
    }
}

File: desktop/src/main/kotlin/com/exodus/game/desktop/DesktopLauncher.kt
package com.exodus.game.desktop

import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3ApplicationConfiguration
import com.exodus.game.ExodusGame

/**
 * Desktop launcher for the Exodus Game.
 * Uses LWJGL3 backend for modern desktop support.
 */
object DesktopLauncher {
    @JvmStatic
    fun main(args: Array<String>) {
        val config = Lwjgl3ApplicationConfiguration().apply {
            // Set window title
            setTitle("Exodus Game")

            // Set window size
            setWindowedMode(1280, 720)

            // Enable vsync for smoother rendering
            setVSync(true)

            // Set foreground FPS limit (0 for unlimited)
            setForegroundFPS(60)

            // Configure MSAA for smoother edges
            setBackBufferConfig(8, 8, 8, 8, 16, 0, 4)
        }

        Lwjgl3Application(ExodusGame(), config)
    }
}
